src/lgdo/lh5/store.py

```diff
@@ -34,7 +36,12 @@ class LH5Store:
     lgdo.waveformtable.WaveformTable
     """
 
-    def __init__(self, base_path: str = "", keep_open: bool = False) -> None:
+    def __init__(
+        self,
+        base_path: str = "",
+        keep_open: bool = False,
+        metacachesize: int = 100,
```

Comment:
It is about 3 MB per file in L200 data (~150 channels * ~50 datasets per channel) and I am storing the metadata at the file level. So it can get very big if you are loading hundreds of files, hence the cap. I think I put more details in the PR comment.

Reply:
Ah! It didn't click that this was the in memory cache -- my bad. Do you think FIFO (vs, say, LRU) is the right policy here?

Reply:
That's a good idea! Do you have some thought on how I should implement that? I am thinking to track the last access time for each file in the metadata cache but maybe there is a better way...

Reply:
So, two thoughts:

1. I suggested LRU as an alternative, but whether that's the better choice depends a lot on the workload. Picking the right cache eviction policy is not easy. That said, I'd _guess_ that LRU is a better choice than FIFO here.
2. I would try as hard as you can to _not_ write this, but look into an existing library. Maybe https://pypi.org/project/pylru/

Also...

I'm concerned about this state being thread safe. It's true that the GIL prevents true _parallelism_, but not _concurrency_. I.e., it's possible to have multiple threads, and without proper locking, an data structure like this could easily fall into an inconsistent state. The library I linked above doesn't have any obvious locking. Potentially we could use this behind an interface and lock all accesses.

See [this discussion](https://stackoverflow.com/questions/105095/are-locks-unnecessary-in-multi-threaded-python-code-because-of-the-gil), e.g.

(I have concerns about the whole library if there is persisted state, but don't know the current state of affairs well enough.)